File: 0066-plus-one/0066-plus-one.swift
class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        var result = digits
        var i = digits.count - 1
        result[i] += 1
        while result[i] > 9 {
            if i == 0 {
                result.insert(1, at: 0)
                i += 1
            } else {
                result[i - 1] += 1
            }
            result[i] = 0
            i -= 1
        }
        return result;
    }
}

File: 0066-plus-one/0066-plus-one.scala
object Solution {
    def plusOne(digits: Array[Int]): Array[Int] = {
        for (i <- digits.length - 1 to 0 by -1) {
          if (digits(i) < 9) {
            digits(i) += 1
            return digits
          }
          digits(i) = 0
        }
        1 +: digits
    }
}